cmd/internal/cli/inspect_linux.go

```diff
@@ -268,121 +392,101 @@ var InspectCmd = &cobra.Command{
 		}
 		name := filepath.Base(abspath)
 
-		a := []string{"/bin/sh", "-c", ""}
-
 		if listApps {
 			sylog.Debugf("Listing all apps in container")
-			a[2] += listAppsCommand
-		}
-
-		// If AppName is given fail quickly (exit) if it doesn't exist
-		if AppName != "" {
-			sylog.Debugf("Inspection of App %s Selected.", AppName)
-			a[2] += getAppCheck(AppName)
+			inspectCMD[2] += listAppsCommand
 		}
 
 		if helpfile {
 			sylog.Debugf("Inspection of helpfile selected.")
-			a[2] += getHelpCommand(AppName)
-		}
-
-		if deffile {
-			sylog.Debugf("Inspection of deffile selected.")
-			a[2] += getDefinitionCommand()
+			inspectCMD[2] += getHelpCommand(AppName)
 		}
 
 		if runscript {
 			sylog.Debugf("Inspection of runscript selected.")
-			a[2] += getRunscriptCommand(AppName)
+			inspectCMD[2] += getRunscriptCommand(AppName)
 		}
 
 		if testfile {
 			sylog.Debugf("Inspection of test selected.")
-			a[2] += getTestCommand(AppName)
+			inspectCMD[2] += getTestCommand(AppName)
 		}
 
 		if environment {
 			sylog.Debugf("Inspection of environment selected.")
-			a[2] += getEnvironmentCommand(AppName)
-		}
-
-		// Default to labels if other flags are unset, excludes --app
-		if labels || defaultToLabels() {
-			sylog.Debugf("Inspection of labels selected.")
-			a[2] += getLabelsCommand(AppName)
-		}
-
-		// Execute the compound command string.
-		fileContents, err := getFileContent(abspath, name, a)
-		if err != nil {
-			sylog.Fatalf("Could not inspect container: %v", err)
+			inspectCMD[2] += getEnvironmentCommand(AppName)
 		}
 
-		inspectObj := inspectFormat{}
-		inspectObj.Type = "container"
-		inspectObj.Attributes.Labels = make(map[string]string)
-		inspectObj.Attributes.Environment = make(map[string]string)
-
-		// Parse the command output string into sections.
-		reader := bufio.NewReader(strings.NewReader(fileContents))
-		for {
-			section, err := reader.ReadBytes('\n')
+		if inspectCMD[2] != "" {
+			// Execute the compound command string.
+			fileContents, err := getFileContent(abspath, name, inspectCMD)
```

Comment:
vars should be defined as close to the usage as possible

Reply:
Its used in the next section, next ~5 lines.

Reply:
Sorry, I meant abspath and name